src/event.rs

```diff
@@ -59,12 +59,76 @@ impl FfmpegOutput {
   }
 }
 
+/// Represents metadata about a stream.
 #[derive(Debug, Clone, PartialEq)]
-pub struct AVStream {
-  /// Typically `video` or `audio`, but might be something else like `data` or `subtitle`.
-  pub stream_type: String,
-  /// Corresponds to stream `-f` parameter, e.g. `rawvideo`, `h264`, or `mpegts`
+pub struct Stream {
+  /// Corresponds to stream `-f` parameter, e.g. `rawvideo`, `h264`, `opus` or `srt`.
   pub format: String,
+  // The language of the stream as a three letter code such as `eng`, `ger` or `jpn`.
+  pub language: String,
+  /// The index of the input or output that this stream belongs to.
+  pub parent_index: usize,
+  /// The index of the stream inside the input.
+  pub stream_index: usize,
+  /// The stderr line that this stream was parsed from.
+  pub raw_log_message: String,
+  // Data that is specific to a certain stream type.
+  pub type_specific_data: TypeSpecificData,
+}
+
+impl Stream {
+  pub fn is_audio(&self) -> bool {
+    matches!(self.type_specific_data, TypeSpecificData::Audio(_))
+  }
+  pub fn is_subtitle(&self) -> bool {
+    matches!(self.type_specific_data, TypeSpecificData::Subtitle())
+  }
+  pub fn is_video(&self) -> bool {
+    matches!(self.type_specific_data, TypeSpecificData::Video(_))
+  }
+  pub fn is_other(&self) -> bool {
+    matches!(self.type_specific_data, TypeSpecificData::Other())
+  }
+
+  pub fn audio_data(&self) -> AudioStream {
+    if let TypeSpecificData::Audio(audio_stream) = &self.type_specific_data {
+      audio_stream.clone()
+    } else {
+      panic!("This is not an audio stream! Check `stream.is_audio()` before calling this method.");
```

Comment:
I'd prefer to avoid an explicit panic here, instead returning an `Option` (or `None` if the stream is of the wrong type).

This would be similar to how `serde_json` handles `Value` types. They have a set of methods like [`as_number()`](https://docs.rs/serde_json/latest/serde_json/enum.Value.html#method.as_number) for the different variants, which is pretty much the same thing we're doing here.

That way it leaves it up to the user to decide whether they want to first check for `is_audio()` and then safely `audio_data().unwrap()`, or use an `if let` on the option directly and handle the error some other way.